src/ORM/Association.php

```diff
@@ -629,9 +629,7 @@ public function deleteAll($conditions)
      */
     protected function _dispatchBeforeFind($query)
     {
-        $table = $this->target();
-        $options = $query->getOptions();
-        $table->dispatchEvent('Model.beforeFind', [$query, new \ArrayObject($options), false]);
+        $query->triggerBeforeFind();
```

Comment:
In my app, I wrote a custom Query class:

    <?php
    namespace App\ORM;
    use Cake\ORM\Query as CakeQuery;

    class Query extends CakeQuery
    {
        public function triggerBeforeFind()
        {
            do_some_stuff();
            parent::triggerBeforeFind();
        }
    }

When retrieving data from a belongsToMany association, cakephp makes 2 select queries. For the second one, the "do_some_stuff()" part on my code was not executed. I just found out it was because it was bypassed by `\Cake\ORM\Association::_dispatchBeforeFind()`.

This is not really a bug (I'm not blaming cakephp since I overwrite part of its code...), but it fixes my problem and make cakephp code DRYer, which is always a good thing :)